### ISSUE TITLE
figure divs for pdf format cause LaTeX error

### ISSUE THREAD

**Issue by kbvernon** [quarto-dev/quarto-cli#4142]
### Bug description

Sorry if this has already been reported. I did a quick search to check, but didn't see anything...

It looks like Pandoc's div blocks for figures, under certain conditions,  generate the LaTeX error - _Not in outer par mode_.

Here's an example:
````md
---
title: "figure divs"
format: pdf

::: {#fig-quarto}
![](quarto.png){fig-align="center" width="2in"}

Caption here
:::
````

Here's what I think is the relevant Quarto output during render:
````md
updating tlmgr

updating existing packages

compilation failed- error
LaTeX Error: Not in outer par mode.
````

A few things to note:
1. This doesn't cause an error for the html format. Everything looks right.
2. If you move the figure options to the div block, as in `::: {#fig-quarto fig-align="center" width="2in"}`, the document renders but the figure options are ignored. Is this expected behavior?
3. Inspecting the .tex file, it appears that adding figure options inside the Pandoc div block, as in the example above, causes Quarto to wrap a LaTeX figure environment inside another figure environment (see below). I kinda wonder if this is what's causing the error, but I don't understand LaTeX sufficiently to say.

````md
\begin{figure}
  {\centering 
    \begin{figure}
      {\centering \includegraphics[width=2in,height=\textheight]{quarto.png}}
    \end{figure}
  }
  \caption{\label{fig-quarto}Caption here}
\end{figure}
````

Here's my system info:
````md
> sessioninfo::session_info()
─ Session info ──────────────────────────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.2.1 (2022-06-23 ucrt)
 os       Windows 10 x64 (build 22621)
 system   x86_64, mingw32
 ui       RStudio
 language (EN)
 collate  English_United States.utf8
 ctype    English_United States.utf8
 tz       America/Denver
 date     2023-01-29
 rstudio  2022.12.0+353 Elsbeth Geranium (desktop)
 pandoc   2.19.2 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)

> quarto::quarto_version()
[1] ‘1.2.329’
````

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
Thanks for the report. Ideally, we'd be more robust in these situations. But I'm wondering if the syntax [we recommend](https://quarto.org/docs/authoring/figures.html#figure-basics) in the docs isn't working for you for some reason.

**Comment by kbvernon** [comment]
Hi, yeah. The standard syntax works:

````md
![Caption here](quarto.png){#fig-quarto fig-align="center" width="2in"}
````

But, when the caption gets longer, it gets a little unwieldy, especially when you want to include text formatting. I suppose you can do this:

````md
![
This is a __long__ caption with _formatting_.
](quarto.png){fig-align="center" width="2in"}
````

though the Pandoc div is still easier to read (mainly because you reference the image before the caption, but it can also handle empty lines, too). I guess that's a minor, somewhat nitpicky point, but it's worth noting that the div syntax works as expected for html, so it seems like it should in the pdf case, too. Or, maybe it should? I would defer to you all on this one.

**Comment by cscheid** [comment]
Eventually we _would_ like to fix things like these, but the fact that this works for HTML is (more or less) a happy accident. The spurious extra div doesn't hurt HTML, but creates a bad scaffolding in .tex that causes the error you're seeing. Leaving this one for the future (it might shake out of some of the changes we're making) but to be honest, there are other matters in the quarto codebase we will want to attend to first.

**Comment by cderv** [comment]
> In fact, the error is caused by the fig-align="center" attribute in the image.

Thanks for the addition !  This is definitely the issue as we are in that case creating two `\begin{figure}` which leads to the error in LaTeX. 

I believe the issue is in that we are processing twice the figure first as a `Div` and then as a `Para` when `fig-align` is set on the figure itself. Code part of Quarto is there. 

https://github.com/quarto-dev/quarto-cli/blob/9c2bcc2f08448048623181768584ecd21c33d96e/src/resources/filters/layout/figures.lua#L6-L41

For each we are creating the logic 

````latex
\begin{figure}

{\centering 
````

And nesting this structure is I think not allowed in LaTeX. We should not process the same when extended figures are inside Div IMO. At least in LaTeX.

**Comment by cscheid** [comment]
This behaves better in the new crossref work, FYI. Keep an eye out for #4944.

**Comment by Robinlovelace** [comment]
Just to add to this, I found the same error message when I added `#| label: ...` labels to subfigures, only affecting pdf outputs.

**Comment by cderv** [comment]
@felipecrp this does work when I try latest 1.4 

````markdown
---
title: "Untitled"
format: pdf
---

::: {#fig-basic}

![Caption for a](surus.png){#fig-basic-a}

![Caption for b](surus.png){#fig-basic-b}

Caption for fig
:::

````

![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/57bb47e4-feae-450e-a934-4fb4d8c2c70d)


The LaTeX produced is 
````latex
\begin{figure}

\centering{

\centering{

\includegraphics{surus.png}

}

\subcaption{\label{fig-basic-a}Caption for a}

\centering{

\includegraphics{surus.png}

}

\subcaption{\label{fig-basic-b}Caption for b}

}

\caption{\label{fig-basic}Caption for fig}

\end{figure}%

````

Please to open a new issue if you have a new issue related to subfigures, and share a reproducible example. 

This issue was about `fig-align` and Figures divs, but I believe this is even fixed. 

````markdown
---
title: "Untitled"
format: pdf
keep-tex: true
---

::: {#fig-basic}

![](surus.png){fig-align='center' width="2in"}

Caption for fig
:::
````

![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/0088af88-8f3c-4148-b0f7-afe2e71b4013)

@cscheid was there anything specific or has this been fixed by the work on Figures for 1.4 ?

**Comment by cscheid** [comment]
It looks fixed to me. If we can get a new reprex for a specific use case, then we should open a new issue. I'm going to go ahead and close.
